homeworks/01-bubble-sort/src/App.tsx

```diff
@@ -0,0 +1,183 @@
+import { ChangeEvent, Component, MouseEvent } from "react";
+
+import Header from "./components/Header";
+import Footer from "./components/Footer";
+import Main from "./components/Main";
+
+import { Col } from "./typings/index";
+
+import generateCols from "./helpers/generate_cols";
+
+import CONFIG from "./config";
+
+import "./App.css";
+
+export enum STATUS {
+  NOT_SOLVED,
+  PAUSED,
+  SORTING,
+  SOLVED,
+}
+
+type AppProps = {};
+
+type AppState = {
+  cols: Array<Col>;
+  status: STATUS;
+  progress: {
+    i: number;
+    j: number;
+  };
+  stepInterval: number;
+  stepByStepMode: boolean;
+  stepTimeout: ReturnType<typeof setTimeout> | number;
+};
+
+class App extends Component<AppProps, AppState> {
+  state = {
+    cols: [],
+    status: STATUS.NOT_SOLVED,
+    progress: {
+      i: 0,
+      j: 0,
+    },
+    stepInterval: CONFIG.SORTING_DELAY,
+    stepByStepMode: true,
+    stepTimeout: 0,
+  };
+
+  componentDidMount(): void {
+    this.resetCols();
+  }
+
+  resetCols(): void {
+    const newCols = generateCols();
+
+    if (this.state.stepTimeout > 0) {
+      clearTimeout(this.state.stepTimeout);
+    }
+
+    this.setState({
+      cols: newCols,
+      status: STATUS.NOT_SOLVED,
+      progress: {
+        i: -1,
+        j: 0,
+      },
+      stepTimeout: 0,
+    });
+  }
+
+  sortingStep(): void {
```

Comment:
Better to keep a business logic separate from the component. E.g., create a pure function and move it to another file (e.g., under `utils` folder).